### PR TITLE
fix(markdown): add missing compilers

### LINKED ISSUE #772: Missing compiler for tableHead.
💂‍♀️ [Sentry](https://sentry.io/organizations/readme/issues/1694192811)
|:---:|

```
Missing compiler for node of type `tableHead`: `[object Object]`
```

### PR DESCRIPTION
**Pull request by rafegoldberg**
💂‍♀️ [Sentry Ticket](https://sentry.io/organizations/readme/issues/1694192811)
|:---:|

## Changes

- [x] Add `tableHead` markdown compiler. Fixes #772. (4d88fb9)
- [x] Add `figure`, `figcaption` compilers. (4d88fb9, 19ad148, 1e12e0c)
- [x] Fix pinned block compiler naming conflict.
- [x] Test figure compiler for magic image blocks. (c6319d6)
- [x] Restructure custom parser and compiler imports. (6e49a17, b49ff91)

### CODE REVIEW

File: packages/markdown/processor/compile/table-head.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = function TableHeadCompiler() {
+  const { Compiler } = this;
+  const { visitors } = Compiler.prototype;
+  visitors.tableHead = function compile(node) {
+    return visitors.tableCell.call(this, node);
+  };
+};
```
Comment (rafegoldberg) [inline packages/markdown/processor/compile/table-head.js:7]: (@gratcliff I wonder if this missing compiler had anything to do with issues re: [indexing table content in Algolia](https://app.asana.com/0/0/1181086456895394/f)?)

Comment (gratcliff) [reply packages/markdown/processor/compile/table-head.js:7]: I think our other table compiler should take care of it. From tabbing through some of their changelogs - they have some super long tables. We're capping individual index records at `2kb`, so they're most likely hitting that cap.

**Comment by gratcliff** [comment]
@rafegoldberg we should talk about our magic block html at some point and how it's being parsed - it currently a little clunky for pulling raw text out.

It's mostly unrelated to this work, but it seems a bit tangential.

File: packages/markdown/processor/compile/table-head.js (hunk 1 above)
Comment (gratcliff) [reply packages/markdown/processor/compile/table-head.js:7]: Following up - it does seem like there's an actual bug. Let me take a look.

Comment (gratcliff) [reply packages/markdown/processor/compile/table-head.js:7]: @rafegoldberg 
[This](https://github.com/readmeio/api-explorer/blob/master/packages/markdown/processor/plugin/table-flattening.js)
seems to be returning joined empty values for that ticket :/

Comment (gratcliff) [reply packages/markdown/processor/compile/table-head.js:7]: Anyway! That's separate work. This looks good.

**Pull request merged**